### PR TITLE
Responsive-Web-Design Milestone 2

### PR DESCRIPTION
**Pull request by danmainah**
MIlestone two involves the continuation of building a https://thenextweb.com/ clone page that is responsive to different screen sizes.
It incorporates the addition of Categories,Latest news, and Latest deals.

### CODE REVIEW

File: index.css [hunk 1]
```diff
@@ -620,21 +906,99 @@ main {
...
 
+  /* Footer section for desktop and tablet view */
   .footer-message {
     flex-direction: row;
   }
 }
 
-@media screen and (min-width: 1024px) {
+@media screen and (min-width: 1025px) {
   main {
```
Comment (Elbie-em) [inline index.css:1071]: This breakpoint should be 1024px

File: index.html [hunk 2]
```diff
@@ -105,6 +107,117 @@ <h2 class="work-sans"><a id="hl-title-txt-side2" href="#">This Mac app automatic
...
+    			</ul>	
+    		</div>	
+		</section>
+
+		<section id="small-news-last">
+
+		</section>		
+
+    </section>	
+
```
Comment (Elbie-em) [inline index.html:221]: Articles in this section should use the article tag, the latest section should be a wrapper generally, same for other sections.

File: index.html [hunk 3]
```diff
@@ -234,19 +347,475 @@ <h4><a class="white-txt work-sans" href="#">Ringover</a></h4>
...
+    				<li><a class="gray-txt-list rajdhani" href="#">CD Projekt Rekt: Cyberpunk 2077 tanks studio's share price nearly 50%</a></li>
+    				<li><hr class="gray-txt-list rajdhani"></li>
+    				<li><a class="gray-txt-list rajdhani" href="#">CD Projekt stock down 20% since Cyberpunk reviews hit the internet</a></li>
+    				<li><hr class="gray-txt-list rajdhani"></li>
+    				<li><a class="gray-txt-list rajdhani" href="#">Airbnb to be bigger than eBay after its long-awaited IPO</a></li>
+    			</ul>	
+			</div>	
+		</section>	
+
+		<section class="news-cont">
```
Comment (Elbie-em) [inline index.html]: Fix duplication in this section as mentioned in main comments.

**Review by Elbie-em — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED 🟡 

Hello @danmainah and @Loosescrew022, you have really worked hard on the project 💪 . You made a great effort meeting almost all the requirements for this milestone. However there some major issues that need to be fixed before this project can be approved.
See my comments within your code, in addition, other improvements indicated below.

## [TO IMPROVE]

- [x] Use of semantic HTML: misuse of sections in your code and makes it very hard for your code to read. Use this [link](https://www.w3schools.com/html/html5_semantic_elements.asp) as a guide on how to make your HTML semantic. 
      - Sections i.e Latest, Categories are the few designated for the *section* tag
      - Articles present in these sections should have an *article* tag
 Other areas could use non-semantic elements, your code compiles fine, but to make sure your page stands out and looks professional, kindly implement semantic practices.

- [x] The desktop version has a huge gap highlights and the Latest News section, please fix this so that it corresponds to the original
Screenshot 2021-04-08 at 13 29 52

- [x] The Latest News section font sizing and element spacing should match the original
- [x] The Lastest News section images do not correspond to the original in your mobile version 
**Yours**
Screenshot 2021-04-08 at 13 37 30
**Original**
Screenshot 2021-04-08 at 13 40 31

- [x] The categories section contains duplicate articles, please fix this to match the original.
**Yours**
Screenshot 2021-04-08 at 13 30 08
**Original**
Screenshot 2021-04-08 at 14 17 31

- [x] The categories section should have two categories per row in the tablet version.
**Yours**
Screenshot 2021-04-08 at 13 46 00
**Original**
Screenshot 2021-04-08 at 14 21 10

- [x] The latest deals section font-sizing and spacing should match the original
- [x] The latest deals section has duplicate titles, please fix this
- [x] The latest deals section images don't match the original
Screenshot 2021-04-08 at 13 51 22.

- [x] There is overflow in mobile and tablet versions, please fix this.

### *Kindly make changes and submit another code review,*
### *All the best, I cant wait to see the final product 😄 *

*As described in the [Code reviews limit Policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 code reviews left for this project.*

*Feel free to contact me for any questions*

> Elbie Moonga (Part-time TSE)

> Github (@Elbie-em)
> Slack (Elbie Moonga)
> Twitter ([@ElbieEm](https://twitter.com/ElbieEm))
> LinkedIn([Elbie Moonga](https://www.linkedin.com/in/elbiemoonga/))


Authors comment;

Hello Elbie! Thank you for your hard work on this review 👍💪. Our team do appreciate that.

We have provided all the requested changes. Below we decided to comment out three of the requested changes to clarify them.

- **Use of semantic HTML** ; Changed 'section' tags in main for 'article' tags. Changed 'section' tags in footer lists for 'nav' tags. Hope that will make code easier to read 😉

- **The desktop version has a huge gap**  ; The gap is coming from a space where you can spot advertise on the live page. If you check Perma.cc record it does have the same gap, that was the reason for us to leave the gap there. Gap has been removed as requested.

- **There is overflow in mobile and tablet versions;** 
Overflow

Could not spot overflow in Github live preview or via local preview. Added margin to the section to make sure it will show up correctly for the reviewer.

File: index.css [hunk 4]
```diff
@@ -358,7 +453,269 @@ main {
...
+
+.news-cont-img6 {
+  height: 35%;
+  width: 100%;
+  background: linear-gradient(rgba(255, 255, 255, 0.2), rgba(0, 0, 0, 1)), url(img/news-img-6.jpg);
+  background-position: center;
+  background-size: cover;
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-end;
```
Comment (Dipeshtwis) [inline index.css]: [Optional]
You have repeated almost all the properties in `.news-cont-imgx` classes where x = 2,3,4,5,6,7,8,9. you can use DRY principle like this. or you can put these properties in one class and use them in HTML.

```
.news-cont-img,
.news-cont-img2,
.news-cont-img3
.news-cont-img4,
......,
 {
  height: 35%;
  width: 100%;
  background-position: center;
  background-size: cover;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  padding: 12px;
}
```

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
# Status:  ♻️ Changes Required ♻️
 Hi @danmainah and @Loosescrew022

**You've fixed almost all the requirements from the previous review. Well done.** Here're the major highlights of this PR:
- Your PR title and description are really descriptive.
- Good use of Semantic HTML.

**However, in order to approve, we have to work a little bit more. Please check my comments below and fix them as appropriate.**


## For Approval

- [x] Still, there is overflow in mobile and tablet versions, please fix this. I have attached the screenshot below.
- The one marked with blue is overlapping with the previous content ( from 320px to 452px).
- The one marked with violet is overflowing if the size goes below 410px. I recommend here using some margin and padding.
- The one marked with blue is starting from the right side due to an overlap issue in almost every screen size. Please fix that.

Screenshot (1310)_LI

## [Optional]

- [x] As this is not mentioned previously. I'm making this optional to fix this. In less than 480px screen size it is only showing two lists instead of 3. ( `Latest funding section`).

Screenshot (1311)_LI

- [x] Use the DRY (Do not repeat yourself) principle wherever possible. I have mentioned some of them in the comments.
- [ ] It would be nice if you create the discount triangle over the top right corner of the image of each article in the `Latest Deals section`.


Remember, This is optional and nothing to do with the project approval.

## Please make the required changes and submit another review request when you are done.

## Reminder

**"As described in the [Code reviews limit Policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) now, you have only 1 code review left for this project. If you think that the code review was not fair, you can [request a second-opinion using this form](https://airtable.com/shrQAqnBwek5a0O0s). "**


(Feel free to contact me on slack by the handle of  **deep** if you have any additional questions).
Happy Coding
[Dipesh Kumar](https://deeptwist.tk)
[Twitter](https://twitter.com/97deepeshkumar)

**Review by DeliceLydia — APPROVED** [review]
# Status: Approved image image

Congratulations @danmainah and @Loosescrew022 for implementing all requested changes this shows your hardworking spirit and readiness to improve 

- Your project is looking good on both mobile, tablet, and desktop devices
- README is descriptive
- PR is descriptive

next time try to group classes with the same properties to apply DRY rules

## Kudos once again feel free to merge and move on to the next project

All the best,

Lydia Ingabire

**Pull request merged**